### ISSUE TITLE
Export default

### ISSUE THREAD

**Issue by jaulz** [viteland/vite-plugin-macro#28]
First of all: great work and thanks for the library! 😊 

While using it I noticed that I cannot export a default (e.g. so I can use `import route from '@macro/route'` instead of `import { route } from '@macro/route'`). I scanned the code but could not see immediately find hints how to achieve it. Any idea if that's considered at all at the moment?

**Comment by jaulz** [comment]
~~Sorry, just the minute after I created the issue I saw that I can~~ ~~use the types parameter and add a custom export~~ ~~there (e.g. `export default route`).~~ ~~That's at least a workaround for the time being~~ 😊 

Edit: obviously this just adds a type but does not really export the function.

**Comment by unbyte** [comment]
Hi could you provide a minimal reproducible example?
I guess you didn't use `{ code: '...' }`?

https://github.com/typed-macro/typed-macro/blob/master/packages/core/README.md#organize-your-macros

**Comment by jaulz** [comment]
Yep, I am using `macros` array to define the macros but I am looking for a way to achieve `export default`. I read the docs quite some time but still don't see a possibility how I can achieve that? I think the macro with the name `default` must be handled different by the library.

**Comment by unbyte** [comment]
I'm afraid I misunderstood your issue.😂
You want to provide a dynamic virtual module with a default export, right? That is, you want the virtual module `'@macros/route'` that macros can change its content?
If then, it's impossible now because in the early stage of this project I found dynamic virtual module is unreliable in building process.
I use macros to insert a route array into code directly. e.g. `mountRoute(yourRouteMacro())`.
And if it cannot meet your need, there are some awesome vite plugins providing virtual module feature, you can try them.

**Comment by jaulz** [comment]
Hm, I think we are still not on the same page 😉 

Currently, I use this to define my macros:
```ts
      createMacroPlugin({
        typesPath: path.join(__dirname, 'types', './macros.d.ts'),
      }).use(
        defineMacroProvider((env) => ({
          id: 'test',
          exports: {
            '@macros/route': {
              macros: [defineMacro(`route`).withHandler(...)],
            },
          },
         })
    )
```

Consequently, the following types are generated:
```ts
declare module '@macros/route' {
  export function route(): void
}
```

I can now import my macro via this:
```ts
import { route } from '@macros/route'
```

However, I would like to use the default export:
```ts
import route from '@macros/route'
```

So it would be great if we could use `defineMacro('default')` to export this type:
```ts
declare module '@macros/route' {
  // Note the default keyword before function and the underscore before the function name (because "default" is a reserved keyword)
  export default function _default(): void
}
```

I hope this was a bit clearer 😊

**Comment by unbyte** [comment]
Currently the default import works the same as `import * as name from '...'`. 
So you can use it like
```
import router from '@macros/router'

router.routes()
```
Your use case is very simple, so I believe that directly using the default export can keep the code concise, but once a lot of macros are written/used in the project, some maintenance problems will arise (to both macro creator and user).
Maybe default exported macros will get supported in the future, but at present, it just matters code style taste and even may lead to code that is not easy to maintain.

**Comment by jaulz** [comment]
Yeah, I think in the end it's a matter of preference but I like to have one function per export. Hence I would like to use:
```ts
import route from '@macro/route'

route()
```

If you like I can give it a try and raise a PR?
